### PR TITLE
fix: stop overwrite tax query filter if it exists

### PR DESCRIPTION
**Pull request by mohjak**
What does this implement/fix? Explain your changes.
---------------------------------------------------
The change is aggregating the taxonomy filters. This is related to [wp-graphql-taxquery](https://github.com/wp-graphql/wp-graphql-tax-query) plugin.

Does this close any currently open issues?
------------------------------------------

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------

Suppose the following scenario:

Given I installed a fresh copy of a WordPress v6.4.3 and WP GraphQL v1.22.0  and WPGraphQL Tax Query
 v0.2.0
And I have created a category with slug 'cat1'
And a tag with slug 'tag1'
And I have created a post that relates only to 'cat1'
When I run the following query:

```graphql
query categoriesSearch {
  categories(where: {slug: "cat1"}) {
    edges {
      node {
        name
        ... on Category {
          posts(
            first: 1
            where: {status: PUBLISH, taxQuery: {relation: AND, taxArray: [{field: NAME, operator: EXISTS, taxonomy: TAG}]}}
          ) {
            nodes {
              title
              status
              tags {
                nodes {
                  name
                }
              }
            }
          }
        }
      }
    }
  }
}
```

Then the following result returns:

```
{
  "data": {
    "categories": {
      "edges": [
        {
          "node": {
            "name": "Cat1",
            "posts": {
              "nodes": [
                {
                  "title": "Post Cat1",
                  "status": "publish",
                  "tags": {
                    "nodes": []
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
```

But, the expected result is to return nothing in posts.

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …
Ubuntu 22.04.4 LTS

**WordPress Version:** …
6.4.3

### CODE REVIEW

**Comment by justlevine** [comment]
Hey @mohjak thanks for opening this PR!

This would be a breaking change to implement, and the same outcome can already be achieved with `get_query_arg()` and `array_merge()`.

I didn't really follow what issue you're trying to solve with this change, but perhaps it makes more sense to do in your custom code with one of the connection resolver hooks, or if it's a bug with the tax query extensions you referenced then a PR to that repo.

( ref: https://github.com/wp-graphql/wp-graphql/issues/3036#issuecomment-1933042912 )

**Comment by mohjak** [comment]
Hi @justlevine,

What I am trying to resolve is the following `set_query_arg` function:

https://github.com/wp-graphql/wp-graphql/blob/9d021498122058854b7b206b406d3deeb5d7bd2c/src/Registry/Utils/TermObject.php#L234

Which is expecting that there is no `tax_query` in the query args that coming from graphql input, but actually it can have a tax_query like my above scenario. So, it overwrites the `tax_query` with the value mentioned in that referenced line of code.

**Comment by justlevine** [comment]
> Hi @justlevine,
> 
> What I am trying to resolve is the following `set_query_arg` function:
> 
> https://github.com/wp-graphql/wp-graphql/blob/9d021498122058854b7b206b406d3deeb5d7bd2c/src/Registry/Utils/TermObject.php#L234
> 
> Which is expecting that there is no `tax_query` in the query args that coming from graphql input, but actually it can have a tax_query like my above scenario. So, it overwrites the `tax_query` with the value mentioned in that referenced line of code.

If anything, seems like it would be the `resolve` you referenced that needs to be "fixed", to check if `tax_query` is set, and if so merge it and pass the results to the `$resolver->set_query_arg()`.

@mohjak can you foresee any issues if we took that (nonbreaking) approach?

File: src/Data/Connection/AbstractConnectionResolver.php [hunk 1]
```diff
@@ -832,7 +832,15 @@ public function get_page_info() {
 	 * @return static
 	 */
 	public function set_query_arg( $key, $value ) {
-		$this->query_args[ $key ] = $value;
+    if ( ! empty( $this->query_args[ $key ] ) && is_array( $this->query_args[ $key ] ) ) {
```
Comment (jasonbahl) [inline src/Data/Connection/AbstractConnectionResolver.php:835]:
```suggestion
		if ( ! empty( $this->query_args[ $key ] ) && is_array( $this->query_args[ $key ] ) ) {
```

**Comment by jasonbahl** [comment]
Apologies for the long-overdue review of this PR. 

I was able to walk through the scenario with the WPGraphQL Tax Query plugin active and reproduce the issue you outlined and I have confirmed that this PR does fix it. 

I wrote up a test based on the scenario that fails without the changes in this PR applied and passes with the changes applied.

**Review by jasonbahl — APPROVED** [review]

**Pull request merged**